src/fancyline/tty/vt100.cr

```diff
@@ -10,11 +24,18 @@ class Fancyline
         super()
       end
 
-      # FIXME: Properly get the terminal dimensions, don't rely on environment
-      #        variables that are outdated at the time of reading them.
+      def winsize
+        LibC.ioctl(0, LibC::TIOCGWINSZ, out winsize)
+        winsize
+      end
 
-      getter columns = ENV["COLUMNS"]?.try(&.to_i?) || 80
-      getter rows = ENV["ROWS"]?.try(&.to_i?) || 25
+      def columns
```

Comment:
If `#winsize` failed, I think it'd be okay for this method to fall back to using `$COLUMNS` and further falling back to `80`.  Hopefully shields against dumb terminals a bit (?).